core/src/impl/Kokkos_HostSpace_deepcopy.cpp

```diff
@@ -43,10 +43,10 @@ void hostspace_parallel_deepcopy_async(void* dst, const void* src,
       "Kokkos::Impl::hostspace_parallel_deepcopy_async: fence after copy");
 }
 
-void hostspace_parallel_deepcopy_async(const DefaultHostExecutionSpace& exec,
-                                       void* dst, const void* src,
-                                       ptrdiff_t n) {
-  using policy_t = Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>;
+template <typename ExecutionSpace>
```

Comment:
Would it make sense to use `template <class ExecutionSpace>` instead of `typename`?

Reply:
He is free to use whichever he likes better, as long as he does not mix in the same template declaration.
We prefer consistency at local scope but we have no strict guideline at the moment.